core/dbt/events/functions.py

```diff
@@ -48,7 +48,7 @@ def setup_event_logger(log_path):
     this.format_color = True if flags.USE_COLORS else False
     # TODO this default should live somewhere better
     log_dest = os.path.join(log_path, 'dbt.log')
-    level = logging.DEBUG if flags.DEBUG else logging.INFO
+    level = logging.DEBUG if flags.DEBUG else this.STDOUT_LOG.level
```

Comment:
If we set it to `WARN` below what's the path for setting it back to `INFO` for runs without the `--debug` flag? I don't want those runs to be stuck on warn instead of info.

Reply:
Yeah, totally fair question. We could set it to `INFO` within the base (non-`list`) task `pre_init_hook`? Perhaps even include the `debug` logic there too?

Ultimately, we'll want to refactor this so that the logger is being instantiated/configured as soon as flags are available, and before all of this other stuff is happening

Reply:
ok so now that I think about it, wouldn't it get reset to info when we call `setup_event_logger` if we DON'T change this line? do we need it to stay warn longer than that?

If so maybe we could change the signature of this function:
```
def setup_event_logger(log_path, level_override=None):
    ...
        level = level_override or logging.DEBUG if flags.DEBUG else logging.INFO
    ...
```

Reply:
@nathaniel-may Just gave this approach a try, I do think it's slightly better